edb/lib/cal.edgeql

```diff
@@ -601,27 +700,71 @@ std::`+` (l: std::duration, r: cal::local_date) -> cal::local_date
     SET commutator := 'std::+';
     SET force_return_cast := true;
     USING SQL $$
-        SELECT ("l" + "r")::edgedb.date_t
+        SELECT ("l" + "r")::edgedb.timestamp_t
     $$;
 };
 
 
 CREATE INFIX OPERATOR
-std::`-` (l: cal::local_date, r: std::duration) -> cal::local_date
+std::`-` (l: cal::local_date, r: std::duration) -> cal::local_datetime
 {
     CREATE ANNOTATION std::identifier := 'minus';
     CREATE ANNOTATION std::description :=
         'Time interval and date/time subtraction.';
     SET volatility := 'Immutable';
     SET force_return_cast := true;
     USING SQL $$
-        SELECT ("l" - "r")::edgedb.date_t
+        SELECT ("l" - "r")::edgedb.timestamp_t
+    $$;
+};
+
+
+CREATE INFIX OPERATOR
+std::`+` (l: cal::local_date, r: cal::relative_duration) -> cal::local_datetime
+{
+    CREATE ANNOTATION std::identifier := 'plus';
+    CREATE ANNOTATION std::description :=
+        'Time interval and date/time addition.';
+    SET volatility := 'Immutable';
+    SET commutator := 'std::+';
+    SET force_return_cast := true;
+    USING SQL $$
+        SELECT ("l" + "r")::edgedb.timestamp_t
+    $$;
+};
+
+
+CREATE INFIX OPERATOR
+std::`+` (l: cal::relative_duration, r: cal::local_date) -> cal::local_datetime
+{
+    CREATE ANNOTATION std::identifier := 'plus';
+    CREATE ANNOTATION std::description :=
+        'Time interval and date/time addition.';
+    SET volatility := 'Immutable';
+    SET commutator := 'std::+';
+    SET force_return_cast := true;
+    USING SQL $$
+        SELECT ("l" + "r")::edgedb.timestamp_t
+    $$;
+};
+
+
+CREATE INFIX OPERATOR
+std::`-` (l: cal::local_date, r: cal::relative_duration) -> cal::local_datetime
```

Comment:
Hm, should we also support multiplication operators for durations? (e.g. date_duration * 2)

Reply:
Maybe. I'd still be inclined to not consider it "essential" and I'm unsure about the correct way of applying `*`. What does "1 month 1 day 1 hour" * 40 produce? Do the days get normalized? Hours? Maybe `duration` just multiplies the elements as is, whereas `relative_duration` normalizes the hours and convers them into days, but leaves elements greater than days as is.

Reply:
OK, let's not implement `*`, `"1 month 1 day 1 hour" * 40` does seem like an example to figure out in the future.